### DISCUSSION TITLE
File System

### DISCUSSION THREAD

**Discussion by NullDev** [shiro-os/shiro-kernel#22] [General]
Settle on a FS and implement it

- EXT4
- BTRFS
- Custom? 

- Already pre-made as librarysuch as [lwext4](https://github.com/gkostka/lwext4) or 
- Written from scratch?

Pro/Cons?

<sub>Linked issue: https://github.com/shiro-os/shiro-kernel/issues/13 </sub>

**Comment by NullDev** [comment]
> EXT4 or a Custom one would make more sense
https://github.com/shiro-os/shiro-kernel/issues/13#issuecomment-831535573

Why wouldn't BTRFS make sense? It's well supported, POSIX Compliant and unlike EXT4 there are open source libs available

**Reply by sharsharshar** [reply]
EXT4 is pretty reliable and easy for users to navigate, and much more stable. But BTRFS is a good option too, I'm just leaning towards ext4 since it's used in almost every other distro.

**Reply by NullDev** [reply]
"almost every other distro" uses Linux. Ext4 is deeply integrated in the linux kernel:

https://github.com/torvalds/linux/tree/master/fs/ext4

which is why it's used. There is no standalone version. We'd either have to use something like lwext4 or implement ext4 from scratch :/

**Reply by sharsharshar** [reply]
Ah I see. Would building our own make sense? Or would that be too strenuous?
